src/interface/blas1_interface.hpp

```diff
@@ -811,14 +811,36 @@ typename sb_handle_t::event_t _rotmg(
   auto d1_view = make_vector_view(_d1, inc, vector_size);
   auto d2_view = make_vector_view(_d2, inc, vector_size);
   auto x1_view = make_vector_view(_x1, inc, vector_size);
-  auto y1_view = make_vector_view(_y1, inc, vector_size);
   auto param_view = make_vector_view(_param, inc, param_size);
 
-  auto operation =
-      Rotmg<decltype(d1_view)>(d1_view, d2_view, x1_view, y1_view, param_view);
-  auto ret = sb_handle.execute(operation, _dependencies);
+  if constexpr (std::is_arithmetic_v<container_3_t>) {
+    constexpr helper::AllocType mem_type = std::is_pointer_v<container_0_t>
+                                               ? helper::AllocType::usm
+                                               : helper::AllocType::buffer;
+    auto _y1_tmp = blas::helper::allocate<mem_type, container_3_t>(
+        1, sb_handle.get_queue());
 
-  return ret;
+    auto copy_y1 = blas::helper::copy_to_device(sb_handle.get_queue(), &_y1,
+                                                _y1_tmp, 1, _dependencies);
+
+    auto y1_view = make_vector_view(_y1_tmp, inc, vector_size);
+    auto operation = Rotmg<decltype(d1_view)>(d1_view, d2_view, x1_view,
+                                              y1_view, param_view);
+#if !defined(AMD_GPU) && !defined(INTEL_GPU) && !defined(NVIDIA_GPU)
+    // This memcpy requires a wait to enforce synchronization for usm. This is
+    // due to workaround a bug present in OpenCL backend that shows up with
+    // portBLAS implementation. Otherwise event dependencies works fine. The
+    // issue has been reported to intel/llvm project here:
+    // https://github.com/intel/llvm/issues/14623
+    if constexpr (mem_type != helper::AllocType::buffer) copy_y1.wait();
```

Comment:
Is this still required since we are not dealing anymore with `host_task` dependencies?

Reply:
Back when  I tested it still seemed necessary, but now with more sync in place it doesn't. I removed in [c128d67](https://github.com/codeplaysoftware/portBLAS/pull/532/commits/c128d670119e99598bf257afc5ebe9caedce28ae). I tested on multiple hardware within a loop and it is fine now. Thanks for pointing it out.